### DISCUSSION TITLE
Diffusion training supporting improvements proposed in literature

### DISCUSSION THREAD

**Discussion by jadevaibhav** [huggingface/diffusers#9572] [Ideas]
With various improvements proposed in diffusion literature, both in training and inference sampling, I think a standard training script to support the new training ideas would be very beneficial. 

I'm looking to implement such training script, initially from changes proposed by: 
1. https://arxiv.org/pdf/2202.00512 (v-prediction and SNR weighting)
2. https://arxiv.org/pdf/2305.08891 (improving on common flaws in noise schedules and sampling)

I am also looking into EDM paper (https://arxiv.org/pdf/2206.00364). 

Do you think such framework would be helpful? Any thoughts and suggestions are most welcome!
@sayakpaul  @patrickvonplaten @asomoza

**Comment by sayakpaul** [comment]
Feel free to do so. https://github.com/crowsonkb/k-diffusion is a good reference.
